.github/workflows/bake.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and publish extensions
+
+on:
+  push:
+
+defaults:
+  run:
+    shell: 'bash -Eeuo pipefail -x {0}'
+
+permissions: {}
+
+jobs:
+  # Gather extensions that have been modified
+  change-triage:
+    name: Check changed files
+    runs-on: ubuntu-24.04
+    outputs:
+      matrix: ${{ steps.get-matrix.outputs.matrix}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5
+
+      - name: Check for changes
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: filter
+        # Remember to add new folders in the operator-changed filter if needed
+        with:
+          base: ${{ (github.event_name == 'schedule') && 'main' || '' }}
+          filters: |
```

Comment:
Can't we generate this from a glob of a checkout in a previous step?  If so, maybe we could filter on directories that have a `metadata.json`.